src/cobra/util/solver.py

```diff
@@ -230,38 +251,42 @@ def get_solver_name(mip=False, qp=False):
         for solver_name in qp_order:
             if solver_name in solvers:
                 return solver_name
-        raise SolverNotFound("no qp-capable solver found")
+        raise SolverNotFound("No QP-capable solver found.")
     else:
         for solver_name in mip_order:
             if solver_name in solvers:
                 return solver_name
-    raise SolverNotFound("no mip-capable solver found")
+    raise SolverNotFound("No MIP-capable solver found.")
 
 
-def choose_solver(model, solver=None, qp=False):
+def choose_solver(
+    model: "Model", solver: Optional[str] = None, qp: bool = False
+) -> Union[ModuleType, Exception]:
```

Comment:
I would not account for exceptions as return values.

```suggestion
) -> ModuleType:
```

Reply:
Okay you have a point there. Will change it. 👍